### PR TITLE
Bug fix 1389239 - Hang due to undefined object when no storage class …

### PR DESCRIPTION
**Pull request by zherman0**
Hang due to undefined object when no storage class is selected

### CODE REVIEW

File: app/scripts/controllers/createPersistentVolumeClaim.js [hunk 1]
```diff
@@ -78,7 +78,7 @@ angular.module('openshiftConsole')
           var unit =  $scope.claim.unit || "Mi";
           pvc.spec.resources.requests.storage = $scope.claim.amount + unit;
           pvc.spec.selector.matchLabels = keyValueEditorUtils.mapEntries( keyValueEditorUtils.compactEntries($scope.claim.selectedLabels) );
-          if ($scope.claim.storageClass !== null) {
+          if ( typeof($scope.claim.storageClass ) !== "undefined" && $scope.claim.storageClass  !== null ) {
```
Comment (spadgett) [inline app/scripts/controllers/createPersistentVolumeClaim.js]: Can we simply check

`if ($scope.claim.storageClass) { ... }`

Comment (zherman0) [inline app/scripts/controllers/createPersistentVolumeClaim.js]: @spadgett - that still does not seem to handle the undefined case which occurs when the user chooses no storage class

Comment (spadgett) [inline app/scripts/controllers/createPersistentVolumeClaim.js]: When `$scope.claim.storageClass` is undefined, that `if` will evaluate to false.

**Comment by zherman0** [comment]
@spadgett - On further testing, that does work for all cases.  Code updated. Thanks.

**Comment by spadgett** [comment]
[merge]

**Pull request merged**